File: IP/3rd/sampleprogram/4-7-3.py
import cv2
import math
import numpy as np
import sys

cap = cv2.VideoCapture(0)  # カメラオープン
if not cap.isOpened():
  print('error')
  sys.exit()

# ウインドウ生成
cv2.namedWindow('src')

# 解像度の設定
cap.set(cv2.CAP_PROP_FRAME_WIDTH, 640)
cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 480)

# 解像度の確認
ret, img_src = cap.read()
print(img_src.shape[1], img_src.shape[0])

while True:
  ret, img_src = cap.read()  # キャプチャ
  cv2.imshow('src', img_src)  # 表示
  key = cv2.waitKey(1)
  if key == ord('q'):
    break

cap.release()
cv2.destroyAllWindows()

File: OpenCVによる画像処理入門/第3版/第2刷/sampleprogram/4-7-3.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;
string win_src = "src";
string win_dst = "dst";

int main()
{
  Mat img_src;
  Mat img_dst;
  VideoCapture capture(0); // カメラオープン
  if(!capture.isOpened()) {
    cout << "error" << endl;
    return -1;
  }

  // ウインドウ生成
  namedWindow(win_src, WINDOW_AUTOSIZE);
  // namedWindow(win_dst, WINDOW_AUTOSIZE);

  // 解像度の設定
  capture.set(CAP_PROP_FRAME_WIDTH, 640);
  capture.set(CAP_PROP_FRAME_HEIGHT, 480);

  // 解像度の確認
  capture >> img_src;
  cout << img_src.cols << " " << img_src.rows << endl;

  while(1) {
    capture >> img_src; // キャプチャ
    imshow(win_src, img_src); // 表示
    if(waitKey(1) == 'q') break;
  }

  capture.release();
  return 0;
}